### PR TITLE
[DOC] Updates to conform to `ruff` `pydocstyle` hook

### PR DESCRIPTION
**Pull request by AnonymousCodes911**
#### Reference Issues/PRs

from: #1191 , See also: #1055 

#### What does this implement/fix? 
`type()` can't be used to check whether an object is of specific type, thus it was replaced by `isinstance()`

#### Does your contribution introduce a new dependency? If yes, which one?
No

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).

##### For developers with write access
- [ ] Optionally, I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst,
Is this what we are aiming to correct? please advice.

**Comment by MatthewMiddlehurst** [comment]
I think at the moment we are only looking to implement the `pydocstyle` like changes i.e. https://github.com/aeon-toolkit/aeon/actions/runs/7933948313/job/21663845905?pr=1055#step:6:172

Feel free to include other code quality improvements though.

**Comment by MatthewMiddlehurst** [comment]
I am fine with these changes, but they are not the suggested documentation ones in the issue. Let me know how you want to proceed.

**Comment by MatthewMiddlehurst** [comment]
@AnonymousCodes911

**Comment by AnonymousCodes911** [comment]
> @AnonymousCodes911

@MatthewMiddlehurst , sorry for missing this, as per

> I think at the moment we are only looking to implement the `pydocstyle` like changes i.e. https://github.com/aeon-toolkit/aeon/actions/runs/7933948313/job/21663845905?pr=1055#step:6:172
> 
> Feel free to include other code quality improvements like the one you made though.

I fixed the superficial code changes that were generating errors when `ruff` was used on the directory, I will try and implement the `pydocstyle` changes as you requested

**Comment by MatthewMiddlehurst** [comment]
If those were also being raised by the default ruff thats fine. The main one if the pydocstyle one though yeah. Dont feel the need to do all them of them as there are quite a lot.

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst, I agree. Since #1196 has been merged, should we also merge and close this one if you think the changes made are fine?

**Comment by MatthewMiddlehurst** [comment]
I mean I don't mind, but the issue it has not really solved any of the items from the linked issue. Please read through them more carefully.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -45,8 +51,12 @@ repos:
     rev: 7.0.0
     hooks:
       - id: flake8
-        additional_dependencies: [ flake8-bugbear, flake8-print, Flake8-pyproject ]
-        args: [ "--max-line-length=88", "--extend-ignore=E203" ]
+        additional_dependencies: [flake8-bugbear, flake8-print, Flake8-pyproject, flake8-docstrings]
+        args:
+          - "--ignore=D103,D107,W503,D300,D400,D200,D202,D401,D101,D205,E704,D100"
```
Comment (AnonymousCodes911) [inline .pre-commit-config.yaml:56]: @MatthewMiddlehurst, I have added the `"--ignore"` flag for the time we are working on these errors. I will remove each error from this list as we resolve it.

**Review by MatthewMiddlehurst — COMMENTED** [review]
I think there may be some confusion about the issue to be resolved. All that needs to happen is to view the functions and file headers currently missing basic docstrings and add them. Some of these changes seem unnecessary, i.e. editing current docstrings.

Perhaps we should just revert the previous commits, merge the changes that were there and focus on another issue i.e. the rocket ones.

**Comment by AnonymousCodes911** [comment]
> I think there may be some confusion about the issue to be resolved. All that needs to happen is to view the functions and file headers currently missing basic docstrings and add them. Some of these changes seem unnecessary, i.e. editing current docstrings.
> 
> Perhaps we should just revert the previous commits, merge the changes that were there and focus on another issue i.e. the rocket ones.

Yes, I didn't merge with the main branch initially before committing these changes. I understand the issue, and if it's alright with you, I would still like to give it a try. I will revert the redundant and unnecessary commits. Sorry for the inconvenience caused.

**Pull request closed without merging**